# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1208373326 14400
#      Wed Apr 16 15:15:26 2008 -0400
# Node ID 9c15f385811c578b8354f389e8af1c948fb12176
# Parent  5b4d278ec828627fc0356ad2d39701d981ab16b0
__gnuplot_version__: use newline at end of message

diff --git a/scripts/plot/__gnuplot_version__.m b/scripts/plot/__gnuplot_version__.m
--- a/scripts/plot/__gnuplot_version__.m
+++ b/scripts/plot/__gnuplot_version__.m
@@ -24,17 +24,21 @@
 
 function version = __gnuplot_version__ ()
 
   persistent __version__ = "";
 
   if (isempty (__version__))
     [status, output] = system (sprintf ("%s --version", gnuplot_binary ()));
     if (status != 0)
-      error ("you must have gnuplot installed to display graphics; if you have gnuplot installed in a non-standard location, see the 'gnuplot_binary' function");
+      ## This message ends in a newline so that the traceback messages
+      ## are skipped and people might actually see the message, read it,
+      ## comprehend it, actually take the advice it gives, and stop
+      ## asking us why plotting fails when gnuplot is not found.
+      error ("you must have gnuplot installed to display graphics; if you have gnuplot installed in a non-standard location, see the 'gnuplot_binary' function\n");
     endif
     pattern = "^[^\\s]*\\s*([0-9]+\\.[0-9]+)\\s*[^\\s]*\\s*([^\\s]*)";
     [d1, d2, d3, d4, matches] = regexp (output, pattern);
     if (iscell (matches) && numel (matches) > 0 && iscellstr (matches{1}))
       __version__ = matches{1}{1};
     endif
   endif
 
